### ISSUE TITLE
请教如何在训练过程中，把box, obj, cls, theta的损失值都打印在屏幕上

### ISSUE THREAD

**Issue by zspzhangshoupeng** [yzqxy/yolov5_obb_prune_tracking#3]
train.py中相关代码有3行

`mloss = torch.zeros(3, device=device)  # mean losses
        #mloss = torch.zeros(4, device=device)  # mean losses`

`LOGGER.info(('\n' + '%10s' * 7) % ('Epoch', 'gpu_mem', 'box', 'obj', 'cls',  'labels', 'img_size'))
        #LOGGER.info(('\n' + '%10s' * 8) % ('Epoch', 'gpu_mem', 'box', 'obj', 'cls', 'theta', 'labels', 'img_size'))`

`pbar.set_description(('%10s' * 2 + '%10.4g' * 5) % (
                #pbar.set_description(('%10s' * 2 + '%10.4g' * 6) % (`

还需要改哪里吗？

**Comment by yzqxy** [comment]
> train.py中相关代码有3行
> 
> `mloss = torch.zeros(3, device=device) # mean losses #mloss = torch.zeros(4, device=device) # mean losses`
> 
> `LOGGER.info(('\n' + '%10s' * 7) % ('Epoch', 'gpu_mem', 'box', 'obj', 'cls', 'labels', 'img_size')) #LOGGER.info(('\n' + '%10s' * 8) % ('Epoch', 'gpu_mem', 'box', 'obj', 'cls', 'theta', 'labels', 'img_size'))`
> 
> `pbar.set_description(('%10s' * 2 + '%10.4g' * 5) % ( #pbar.set_description(('%10s' * 2 + '%10.4g' * 6) % (`
> 
> 还需要改哪里吗？

训练过程中，是会自动打印训练的l各部分loss的

**Comment by zspzhangshoupeng** [comment]
> > train.py中相关代码有3行
> > `mloss = torch.zeros(3, device=device) # mean losses #mloss = torch.zeros(4, device=device) # mean losses`
> > `LOGGER.info(('\n' + '%10s' * 7) % ('Epoch', 'gpu_mem', 'box', 'obj', 'cls', 'labels', 'img_size')) #LOGGER.info(('\n' + '%10s' * 8) % ('Epoch', 'gpu_mem', 'box', 'obj', 'cls', 'theta', 'labels', 'img_size'))`
> > `pbar.set_description(('%10s' * 2 + '%10.4g' * 5) % ( #pbar.set_description(('%10s' * 2 + '%10.4g' * 6) % (`
> > 还需要改哪里吗？
> 
> 训练过程中，是会自动打印训练的l各部分loss的

并不是，theta就没有打印。

**Comment by yzqxy** [comment]
> > > train.py中相关代码有3行
> > > `mloss = torch.zeros(3, device=device) # mean losses #mloss = torch.zeros(4, device=device) # mean losses`
> > > `LOGGER.info(('\n' + '%10s' * 7) % ('Epoch', 'gpu_mem', 'box', 'obj', 'cls', 'labels', 'img_size')) #LOGGER.info(('\n' + '%10s' * 8) % ('Epoch', 'gpu_mem', 'box', 'obj', 'cls', 'theta', 'labels', 'img_size'))`
> > > `pbar.set_description(('%10s' * 2 + '%10.4g' * 5) % ( #pbar.set_description(('%10s' * 2 + '%10.4g' * 6) % (`
> > > 还需要改哪里吗？
> > 
> > 
> > 训练过程中，是会自动打印训练的l各部分loss的
> 
> 并不是，theta就没有打印。

box loss包括theta，并不是像原版把theta单独计算loss

**Comment by zspzhangshoupeng** [comment]
大佬，你有没有本项工作的相关论文，以后可能需要引用以下。

**Comment by yzqxy** [comment]
> 大佬，你有没有本项工作的相关论文，以后可能需要引用以下。

目前还没考虑写，有你需要的你用就行，记得给我点个star
